tests/test_acceptance.py

```diff
@@ -258,16 +294,37 @@ class Meta:
         self.assertRaises(ValidationError, s2.validate)
 
     def test_missing_fields(self):
-        class Obj(object):
+        class DefaultModelObj(object):
             def __init__(self):
                 self.string_field = "strfield"
 
         class ObjSerializer(Serializer):
             class Meta:
-                model = Obj
+                model = DefaultModelObj
 
             string_field = StringField(name="stringField", required=False, allow_null=False)
 
         s = ObjSerializer(data={})
         s.validate()
-        self.assertEqual(s.object.string_field, Obj().string_field)
+        self.assertEqual(s.object.string_field, DefaultModelObj().string_field)
+
+        # object to data
+        object_with_missing_field = type('DefaultModelObj', (object,), s.data)
+        deserializer = ObjSerializer(object=object_with_missing_field)
+        deserializer.validate()
+        self.assertEqual(deserializer.data['stringField'], DefaultModelObj().string_field)
+
+    def test_missing_fields_without_default_model(self):
+        class ObjSerializer(Serializer):
+            string_field = StringField(name="stringField", required=False, allow_null=False)
+
+        data = {}
+        s = ObjSerializer(data=data)
+        s.validate()
+        self.assertEqual(getattr(s.object, 'string_field', None), data.get('StringField'))
```

Comment:
So it seems the assertion here was a bit confusing. To be clear the intended functionality is this.

When serializing data without fields defined in the serializer - if the field in the serializer is not required AND no default model is set then the serializer should pass validation while serializing that data. 

If I were to change the assertion to be more clear I would replace it with something like this.

self.assertFalse(hasattr(s.object, 'string_field))